test/inductor/test_inductor_freezing.py

```diff
@@ -375,6 +471,139 @@ def foo(mod, x):
+                stride=2,
+                running_mean=running_mean,
+                running_var=running_var,
+                weight=torch.ones(32).to(self.device),
+                bn_bias=torch.zeros(32).to(self.device),
+            )
+            .eval()
+            .to(self.device)
+            .to(torch.float32)
+        )
+
+        def foo(mod, x):
+            mod(x)
+            return mod(x)
+
+        with torch.no_grad():
+            out_eager = foo(mod, x)
+            out_optimized_for_infernece, code = run_and_get_code(
+                torch.compile(foo), mod, x
+            )
+
+        self.assertNotIn(
+            "aten._native_batch_norm_legit_no_training(",
+            code[0],
+        )
+
+        self.assertEqual(out_optimized_for_infernece, out_eager, atol=1e-2, rtol=1e-2)
+
+    @torch._inductor.config.patch(layout_optimization=False)
+    def test_conv_bn_with_multi_bn_share_conv(self):
+        mod = (
+            ConvMultiBN(32, 32, bias=True, kernel_size=3, stride=2)
+            .to(self.device)
+            .to(torch.float32)
+        )
+
+        # Update the default parameters of BN module
+        for _ in range(10):
+            mod(torch.rand(3, 32, 32, 32).to(self.device).to(torch.float32))
+
+        mod.eval()
+        x = torch.rand(3, 32, 32, 32).to(self.device).to(torch.float32)
+
+        def foo(mod, x):
+            return mod(x)
+
+        with torch.no_grad():
+            out_eager = foo(mod, x)
+            out_optimized_for_infernece, _ = run_and_get_code(
+                torch.compile(foo), mod, x
+            )
+
+        self.assertEqual(out_optimized_for_infernece, out_eager, atol=1e-2, rtol=1e-2)
+
+    @torch._inductor.config.patch(layout_optimization=False)
+    def test_conv_functional_bn_with_multi_bn_share_conv(self):
+        x = torch.rand(3, 32, 32, 32).to(self.device).to(torch.float32)
+        running_mean = torch.mean(x, dim=(0, 2, 3)).to(self.device)
+        running_var = torch.var(x, dim=(0, 2, 3)).to(self.device)
+        running_mean2 = torch.mean(x, dim=(0, 2, 3)).to(self.device)
+
+        mod = (
+            ConvMultiFunctionalBN(
+                32,
+                32,
+                bias=True,
+                kernel_size=3,
+                stride=2,
+                running_mean=running_mean,
+                running_var=running_var,
+                weight=torch.ones(32).to(self.device),
+                bn_bias=torch.zeros(32).to(self.device),
+                running_mean2=running_mean2,
+            )
+            .eval()
+            .to(self.device)
+            .to(torch.float32)
+        )
+
+        def foo(mod, x):
+            return mod(x)
+
+        with torch.no_grad():
+            out_eager = foo(mod, x)
+            out_optimized_for_infernece, _ = run_and_get_code(
+                torch.compile(foo), mod, x
+            )
+        self.assertEqual(out_optimized_for_infernece, out_eager, atol=1e-2, rtol=1e-2)
```

Comment:
Shall we also check if BN still exists in the graph?

Reply:
Actually, the BN has been decomposed in https://github.com/pytorch/pytorch/blob/57790fd088af40d9ab44c66def34db23f9d2879f/torch/_decomp/decompositions.py#L1704. Remove related checkers.